lua/apisix/api/init.lua

```diff
@@ -0,0 +1,103 @@
+local core = require("apisix.core")
+local route = require("resty.r3")
+local get_method = ngx.req.get_method
+local str_lower = string.lower
+local ngx = ngx
+
+
+local resources = {
+    config    = require("apisix.api.config"),
+}
+
+
+local _M = {version = 0.1}
+local router
+
+
+local function check_signature(signature, timestamp, res)
+    if not signature then
+        return nil, "api signature is needed"
+    end
+
+    local now = os.time()
+    if now - timestamp > 5 then
+        return nil, "api signature expired"
+    end
+
+    local local_conf = core.config.local_conf()
+
+    local src        = res..'|'..timestamp
+    local digest     = ngx.hmac_sha1(local_conf.apisix.api_key, src)
+    local str_base64 = ngx.encode_base64(digest)
+    local generated  = ngx.md5(str_base64)
+
+    if generated==signature then
+        return true, nil
+    end
+
+    return nil, "signature checked fail"
+end
+
+
+local function run(params)
+
+    local resource = resources[params.res]
+    if not resource then
+        core.response.exit(404)
+    end
+
+    local method = str_lower(get_method())
+    if not resource[method] then
+        core.response.exit(404)
+    end
+
+    ngx.req.read_body()
+    local req_body = ngx.req.get_body_data()
```

Comment:
https://github.com/openresty/lua-nginx-module#ngxreqget_body_data

This function returns nil if

* the request body has not been read,
* the request body has been read into disk temporary files,
* or the request body has zero size.